ui-components/src/page_components/listing/ListingCard.tsx

```diff
@@ -22,10 +26,80 @@ export interface ListingCardProps {
   tableHeaderProps?: ListingCardHeaderProps
   tableProps: ListingCardTableProps
   detailsLinkClass?: string
+  listingID: string
 }
 
 const ListingCard = (props: ListingCardProps) => {
   const { imageCardProps, tableProps, detailsLinkClass, tableHeaderProps } = props
+  const { profile, userProfileService } = useContext(AuthContext)
+  const preferences: UserPreferences = profile?.preferences || {
+    sendEmailNotifications: false,
+    sendSmsNotifications: false,
+    favoriteIDs: [],
+  }
+  const [favoriteState, updateFavoriteState] = useState("favored")
```

Comment:
Why is this a string called "favored"? Do you just want a boolean here?

Reply:
updated to bool.  Had string just for readability.